cli/data.go

```diff
@@ -595,6 +595,131 @@ func (c *viamClient) dataAddToDataset(datasetID, orgID, locationID string, fileI
 	return nil
 }
 
+// DataAddToDatasetByFilter is the corresponding action for 'data dataset add filter'.
+func DataAddToDatasetByFilter(c *cli.Context) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+	filter, err := createDataFilter(c)
+	if err != nil {
+		return err
+	}
+	if err := client.dataAddToDatasetByFilter(filter, c.String(datasetFlagDatasetID)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func performActionOnBinaryDataFromFilter(actionOnBinaryData func(*datapb.BinaryID) error,
```

Comment:
This could be an unexported helper on viamClient so that you don't have to pass in the dataClient or Writer.

Reply:
Done